spec/lang/operator/or_spec.lua

```diff
@@ -191,6 +191,10 @@ describe("or", function()
       local _print_testcty: nil = _testc
       local _print_testdty: nil = _testd
    ]], {
+      { y = 6, msg = "the resulting type is ambiguous: A or B" },
```

Comment:
Ohhh, I was wondering how we were ending up with an ambiguous type here -- in other words, I understand why `A :> B` (`rb is ra`) but it wasn't immediately clear to me why `B :> A`... 

Looks like the case `interface is record` is being matched _structurally_, which is odd! I don't remember why I ever put that in.

I just tried simply deleting the `["interface"]["record"]` rule from `subtype_relations`, and it actually picked up a couple of type declaration bugs in the compiler code (and the test suite passed otherwise).  I'll open a PR with that change!

Reply:
Please check out #980 !